app/models/assignment_properties.rb

```diff
@@ -51,6 +51,47 @@ class AssignmentProperties < ApplicationRecord
 
   validate :minimum_number_of_groups
 
+  attribute :duration, :duration
```

Comment:
This is interesting, what is the purpose of this line?

Reply:
This tells the model that the `duration` attribute (which is stored as a string in the database) should be interpreted as a [duration](https://api.rubyonrails.org/classes/ActiveSupport/Duration.html) data type, which we have registered previously in `config/initializers/types.rb` for use with the `extensions` model.